spec/unit/room.spec.ts

```diff
@@ -174,19 +174,19 @@ describe("Room", function () {
         tsThread?: number,
     ): { mainEvent?: MatrixEvent; threadEvent?: MatrixEvent } => {
         const result: { mainEvent?: MatrixEvent; threadEvent?: MatrixEvent } = {};
+        const { rootEvent, thread } = mkThread({
+            room,
+            client: new TestClient().client,
+            authorId: "@bob:example.org",
+            participantUserIds: ["@bob:example.org"],
+        });
```

Comment:
It seems to me that adding this here means that we are no longer testing the behaviour when there are no thread events? Or is that tested somewhere else?

I'm unclear what exactly `addRoomMainAndThreadMessages` is supposed to do, and what the semantics of `tsMain` and `tsThread` are supposed to do. Given the fact that we need to change this, I'm evidently not alone.

Please could you give the function a proper doc comment describing what it does, and what `tsMain` and `tsThread` do.

Reply:
@richvdh take a look at the last commit. I threw away the magic setup function. It should (hopefully) be clearer now.